src/cli/init.js

```diff
@@ -0,0 +1,183 @@
+                           [--print]
+           sourcecred init --help
+
+    Sets up a new SourceCred instance, by creating a SourceCred project
+    configuration, and saving it to 'sourcecred.json' within the current
+    directory.
+
+    Zero or more GitHub specs may be provided; each GitHub spec can be of the
+    form OWNER/NAME (as in 'torvalds/linux') for loading a single repository,
+    or @owner (as in '@torvalds') for loading all repositories owned by a given
+    account. If any GitHub specs are present, then the SOURCECRED_GITHUB_TOKEN
+    environment variable is required.
+
+    A discourse url may be provided. The discourse url must be the full url of
+    a valid Discourse server, as in 'https://discourse.sourcecred.io'.
+
+    All of the GitHub specs, and the Discourse specification (if it exists)
+    will be combined into a single project, which is written to
+    sourcecred.json. The file may be manually modified to activate other
+    advanced features, such as identity map resolution.
+
+    Arguments:
+        --github GITHUB_SPEC
+            A specification (in form 'OWNER/NAME' or '@OWNER') of GitHub
+            repositories to load.
+
+        --discourse DISCOURSE_URL
+            The url of a Discourse server to load.
+
+        --force
+            If provided, sourcecred init will overwrite pre-existing
+            sourcecred.json files. Otherwise, the command will refuse to
+            overwite pre-existing files and fail.
+
+        --print
+            If provided, sourcecred init will print the project to stdout
+            rather than writing it to sourcecred.json. Mostly used for testing
+            purposes, as a SourceCred instance is only valid if the file is
+            saved as sourcecred.json. If this flag is set, it supercedes
+            --force.
+
+        --help
+            Show this help message and exit, as 'sourcecred help init'.
+
+    Environment variables:
+        SOURCECRED_GITHUB_TOKEN
+            API token for GitHub. This should be a 40-character hex
+            string. Required if you provide GitHub specs.
+
+            To generate a token, create a "Personal access token" at
+            <https://github.com/settings/tokens>. When loading data for
+            public repositories, no special permissions are required.
+            For private repositories, the 'repo' scope is required.
+    `.trimRight()
+  );
+}
+
+function die(std, message) {
+  std.err("fatal: " + message);
+  std.err("fatal: run 'sourcecred help init' for help");
+  return 1;
+}
+
+const initCommand: Command = async (args, std) => {
+  let withForce = false;
+  let printToStdOut = false;
+  let discourseUrl: ?string;
+  let githubSpecs: string[] = [];
+
+  for (let i = 0; i < args.length; i++) {
+    switch (args[i]) {
+      case "--help": {
+        usage(std.out);
+        return 0;
+      }
+      case "--github": {
+        if (++i >= args.length) return die(std, "--github given without value");
```

Comment:
The `(++i >= args.length)` check seems off to me. Aside from code golfing, what happens with:
`sourcecred init --github --github sourcecred/sourcecred`?

Reply:
What seems off to you about the check?

`sourcecred init --github --github sourcecred/sourcecred` will error:

`fatal: unexpected argument sourcecred/sourcecred`

Providing the github flag twice in a row is an invalid input, so that it errors is fine. If we wanted an even more helpful error message, we could do validation on each argument to `--github` to check that it looks like a valid github spec, at which point it would say `fatal: "--github" is not a valid github spec`, which is perhaps more helpful.